### PR TITLE
Replace Model.java with model.c from stats package

### PR DESCRIPTION
**Pull request by psolaimani**

### CODE REVIEW

**Comment by akbertram** [comment]
Looking good!

Can we also remove the old java implementations?

File: packages/stats/src/main/c/model.c [hunk 1]
```diff
@@ -787,7 +787,8 @@ static SEXP ExpandDots(SEXP object, SEXP value)
 	    }
 	    else if (length(object) == 3) {
 		SETCADR(object, ExpandDots(CADR(object), value));
-		SETCADDR(object, ExpandDots(CADDR(object), value));
+		SETCADDR(object, lang2(parenSymbol,
```
Comment (akbertram) [inline packages/stats/src/main/c/model.c]: Why is this necessary?

Comment (psolaimani) [reply packages/stats/src/main/c/model.c]: In Renjin, the way parenthesis are added works correctly for expanding the dot `.` for `*` interaction but not for `+` 
for example
```
df <- data.frame(x=1:3, y=1:3, z=1:3)
f <-  x ~ y * .
terms(f, data = df)
```
gives **x ~ y * (y + z)**

but 
```
df <- data.frame(x=1:3, y=1:3, z=1:3)
f <-  x ~ y + .
terms(f, data = df)
```
gives **x ~ y + y + z** instead of **x ~ y + (y + z)**

in [ExpandDots()](https://github.com/wch/r-source/blob/tags/R-3-4-0/src/library/stats/src/model.c#L771) function, for [* case](https://github.com/wch/r-source/blob/tags/R-3-4-0/src/library/stats/src/model.c#L820) parenthesis are [added explicitly](https://github.com/wch/r-source/blob/tags/R-3-4-0/src/library/stats/src/model.c#L831). So I added the same logic for the `+ case` above. 

The reason why I didn't merge is because I still didn't get why/how in GNU R this parenthesis are added to the `+ case`. The input to `ExpandDots()` is exactly the same for Renjin and GNU R, but the output of Renjin was missing the parenthesis. I couldn't detect the program control flow difference in `ExpandDots()`

Comment (akbertram) [reply packages/stats/src/main/c/model.c]: But the C code for ExpandDots is identical in both GNU R and Renjin, now, or not?

Comment (psolaimani) [reply packages/stats/src/main/c/model.c]: Now its not identical anymore.

When expanding the `.`, parenthesis are added for the `+` case explicitly  (`lang2(parenSymbol, ... )`) instead of magically.

**Comment by psolaimani** [comment]
merged manually with this [commit](https://github.com/bedatadriven/renjin/commit/84b208f98dfebedd244b59f5f179a4f237a7c0bf)
